rust/src/fvm/machine.rs

```diff
@@ -147,8 +151,29 @@ pub unsafe extern "C" fn fil_create_fvm_machine(
         let blockstore = blockstore.finish();
 
         let externs = CgoExterns::new(externs_id);
+
+        let engine = match (|| -> anyhow::Result<Engine> {
+            let mut engines = ENGINES
+                .lock()
+                .map_err(|_| anyhow::Error::msg("engines lock is poisoned"))?;
```

Comment:
`anyhow::anyhow!` is the more common way of doing this, especially including the error would be nice if you do this